### ISSUE TITLE
RFE: coords.format precision with python 3

### ISSUE THREAD

**Issue by kglotfelty** [cxcsds/ciao-contrib#123]
https://github.com/cxcsds/ciao-contrib/blob/7004e9ae1f846aea102b7665415765e50267065b/ciao-4.10/contrib/lib/python3.5/site-packages/coords/format.py#L844-L845

The built-in test for the coord/formater fails with python 3 due to the increased numerical precision.  With ciao 4.10 p3 (Linux) I'm getting:

```bash
% python format.py 
..FF..F.F
======================================================================
FAIL: test_deg2dec (__main__.TestModule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "format.py", line 792, in test_deg2dec
    self._test_expected_dec)
  File "format.py", line 748, in _test_fmtconvert
    msg="format='{}'".format(fmt))
AssertionError: '-16d 42\' 47.304"' != '-16d 42\' 47.30399999999719"'
- -16d 42' 47.304"
+ -16d 42' 47.30399999999719"
 : format='dms'

======================================================================
FAIL: test_deg2ra (__main__.TestModule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "format.py", line 787, in test_deg2ra
    self._test_expected_ra)
  File "format.py", line 748, in _test_fmtconvert
    msg="format='{}'".format(fmt))
AssertionError: '6H 45M 9.2496S' != '6H 45M 9.249599999997713S'
- 6H 45M 9.2496S
+ 6H 45M 9.249599999997713S
 : format='HMS'

======================================================================
FAIL: test_ra2deg (__main__.TestModule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "format.py", line 797, in test_ra2deg
    self._test_expected_ra)
  File "format.py", line 756, in _test_convert
    msg="val='{}'".format(val))
AssertionError: '101.28854' != '101.28853999995'
- 101.28854
+ 101.28853999995
 : val='6.75256933333h'

======================================================================
FAIL: test_small_neg_dec (__main__.TestModule)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "format.py", line 809, in test_small_neg_dec
    self.assertEqual(out, strval, msg='to string')
AssertionError: '-0:0:5.0000000000039995' != '-0:0:5.0'
- -0:0:5.0000000000039995
+ -0:0:5.0
 : to string

----------------------------------------------------------------------
Ran 9 tests in 0.007s

FAILED (failures=4)
```

This also causes failures in some tests of the contrib scripts: all `merge_obs` tests fail in some way because of it.  (The precision is enough to make re-projections just a little different, image sizes different, etc.)

Is there a way to sensibly round these values ?

**Comment by DougBurke** [comment]
When I last looked at this it wasn't obvious to me to how best to handle this. Any thoughts welcome.

**Comment by DougBurke** [comment]
This is partially addressed by #420 - which adds an optional `ndp` argument to limit the number of decimal places

**Comment by DougBurke** [comment]
#422 "fixes" the tests, at the expense of requiring `pytest`
